test/functional/test_framework/test_framework.py

```diff
@@ -403,7 +403,8 @@ def init_wallet(self, i):
             n = self.nodes[i]
             if wallet_name is not None:
                 n.createwallet(wallet_name=wallet_name, descriptors=self.options.descriptors, load_on_startup=True)
-            n.importprivkey(privkey=n.get_deterministic_priv_key().key, label='coinbase')
+            if self.chain != "main":
```

Comment:
Why?

Reply:
because all the key address pair we have now are only for testnets 
and will when the chain get initialized used but fail checks, so that the test
would not even start,
We can gen them on the fly if we need keys and addresses. For mainchain tests.?